vnext/Microsoft.ReactNative/Views/Image/ReactImage.h

```diff
@@ -29,13 +31,19 @@ struct ReactImageSource {
   ImageSourceFormat sourceFormat = ImageSourceFormat::Bitmap;
 };
 
+using ImageCache = typename LRUCache<std::string, xaml::Media::Imaging::BitmapImage, 100>;
+using SurfaceCache = typename LRUCache<std::string, xaml::Media::LoadedImageSurface, 100>;
```

Comment:
In a worst case real life scenario, the framework could see back to back high resolution images. A 4K ARGB bitmap is ~32MB. In this scenario when high resolution images are loaded back to back, it seems like this cache could grow into the GBs.

Could we put a MB-based cap on the cache? This may be different for different device classes (e.g. systems with 4GB total memory). If we do end up with a very large cache, it would be good to have a strategy to free resources when appropriate. E.g. responding to system low-memory events, or proactively evicting "old" resources.

Reply:
Good point. We do need a real memory cache that is able to set a cap on the memory limit on a single file. This PR is meant to provide a stop gap to serve the most common cases. 
I'd prefer that we leave that real memory cache to another PR.

Reply:
I am no longer a maintainer of the repo. It looks like @acoates-ms signed off on the PR, but I think someone at MSFT has to hit the merge button, once it has been reviewed by the appropriate folks.

I think it's reasonable as a follow-up, but we also have a "QuirkSettings" struct that we use for feature-gating. That could allow keeping the change off by default, while it is still being developed.

Out of curiosity, is this PR to upstream a change that already exists in a Meta fork of RNW? Or is it being developed here?

Reply:
the change has already existed in the Meta fork. this PR is to upstream to RNW.

Reply:
Yeah, there is some concern about the potential memory usage of this if we turn it on for everyone right now.  Can we put this behind a QuirkSetting for now?  That way we can do better AB testing and enable the feature to mature a little before we turn it on for everyone.